iothub/device/src/InternalClient.cs

```diff
@@ -185,9 +165,28 @@ public async Task OpenAsync(CancellationToken cancellationToken = default)
         }
 
         /// <summary>
-        /// Sends an event to device hub
+        /// Sends an event to IoT hub. The client instance must be opened already.
         /// </summary>
-        /// <returns>The message containing the event</returns>
+        /// <remarks>
+        /// In case of a transient issue, retrying the operation should work. In case of a non-transient issue, inspect
+        /// the error details and take steps accordingly.
+        /// Please note that the list of exceptions is not exhaustive.
+        /// </remarks>
+        /// <param name="message">The message to send.</param>
+        /// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
+        /// <exception cref="ArgumentNullException">Thrown when a required parameter is null.</exception>
+        /// <exception cref="IotHubClientException">Thrown and <see cref="IotHubClientException.StatusCode"/> is set to <see cref="IotHubStatusCode.NetworkErrors"/>
+        /// when the operation has been canceled. The inner exception will be <see cref="OperationCanceledException"/>.</exception>
```

Comment:
@brycewang-microsoft you'll be changing this to throw OperationCanceledException natively instead of as an inner exception, right?